tensorflow/compiler/xla/service/gpu/gpu_fusible.cc

```diff
@@ -133,17 +133,18 @@ bool ShapesCompatibleForMultiOutputFusion(const HloInstruction& instr1,
 
 bool IsInputFusibleScatter(const HloInstruction& instr) {
   if (instr.opcode() == HloOpcode::kScatter ||
-     (instr.opcode() == HloOpcode::kFusion &&
-      instr.fusion_kind() == HloInstruction::FusionKind::kInput &&
-      instr.fused_expression_root()->opcode() == HloOpcode::kScatter)) {
+      (instr.opcode() == HloOpcode::kFusion &&
+       instr.fusion_kind() == HloInstruction::FusionKind::kInput &&
+       instr.fused_expression_root()->opcode() == HloOpcode::kScatter)) {
     return true;
   }
   return false;
 }
 
 bool IsInputFusible(const HloInstruction& instr) {
   // Input fusion only handles non-elemental reduction and scatter operations.
-  return IsInputFusibleReduction(instr) || IsInputFusibleScatter(instr);
+  return instr.IsFusible() &&
```

Comment:
Looks like you made the same changes in PR 26277. That makes sense, no need for a separate PR in my opinion.